src/uwtools/rocoto.py

```diff
@@ -25,20 +35,118 @@ def _add_jobname(tree: dict) -> None:
             _add_jobname(subtree)
 
 
-# Public functions
-def write_rocoto_xml(input_yaml: str, input_template: str, rendered_output: str) -> None:
+def _add_jobname_to_tasks(
+    input_yaml: OptionalPath = None,
+) -> YAMLConfig:
     """
-    Main entry point.
+    Load YAML config and add job names to each defined workflow task.
 
     :param input_yaml: Path to YAML input file.
-    :param input_template: Path to input template file.
-    :param rendered_output: Path to write rendered XML file.
     """
     values = YAMLConfig(input_yaml)
-    tasks = values["tasks"]
+    tasks = values["workflow"]["tasks"]
     if isinstance(tasks, dict):
         _add_jobname(tasks)
+    return values
+
+
+def _rocoto_schema_xml() -> DefinitePath:
+    """
+    The path to the file containing the schema to validate the XML file against.
+    """
+    with resources.as_file(resources.files("uwtools.resources")) as path:
+        return path / "schema_with_metatasks.rng"
+
+
+def _rocoto_schema_yaml() -> DefinitePath:
+    """
+    The path to the file containing the schema to validate the YAML file against.
+    """
+    with resources.as_file(resources.files("uwtools.resources")) as path:
+        return path / "rocoto.jsonschema"
+
+
+def _rocoto_template_xml() -> DefinitePath:
+    """
+    The path to the file containing the Rocoto workflow document template to render.
+    """
+    with resources.as_file(resources.files("uwtools.resources")) as path:
+        return path / "rocoto.jinja2"
+
+
+def _write_rocoto_xml(
+    config_file: OptionalPath,
+    rendered_output: DefinitePath,
+) -> None:
+    """
+    Render the Rocoto workflow defined in the given YAML to XML.
+
+    :param config_file: Path to YAML input file.
+    :param rendered_output: Path to write rendered XML file.
+    """
+
+    values = _add_jobname_to_tasks(config_file)
```

Comment:
I'm not sure on this one, but... The only call to `_write_rocoto_xml()` I see is via `_realize_rocoto_xml()`, which itself calls `_add_jobname_to_tasks()`. So, presumably the workflow YAML has already been updated with job names by the time we get here. Do we need this call? It probably does no harm, but it seems potentially redundant, and so possibly confusing.

Reply:
Oh, true, we should only need it at the `_write()`, this one should be able to go

Reply:
It's possible that, when unit-testing this function directly, you might need to call `_add_jobname_to_tasks()` in the test if that's necessary for it to pass. Maybe it's not, though.